### ISSUE TITLE
Crash when importing a mod with Mixin Squared

### ISSUE THREAD

**Issue by Crystal-Spider** [Bawnorton/MixinSquared#13]
I have a mod of mine, let's call it A, that uses Mixin Squared in the fabric module.  
In that mod, the dependency is added with the line `include(implementation(annotationProcessor("com.github.bawnorton.mixinsquared:mixinsquared-fabric:0.2.0-beta.4")))`.

A problem I faced initially was that my other mod, let's call it B, that depends on this one had to add the jitpack repository too, otherwise it would crash during build saying that mixinsquared couldn't be found anywhere.  
Is there a way to avoid forcing other mods to add jitpack repository? I'd like for my mod A to be imported in B without the need to do extra stuff.

A second, more blocking, problem is that both A and B crash at runtime with the following error:
```
[10:06:42] [main/ERROR] (FabricLoader) Minecraft has crashed!
 net.fabricmc.loader.impl.FormattedException: java.lang.RuntimeException: Mixin transformation of net.minecraft.world.entity.player.Player failed
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:463) ~[fabric-loader-0.14.14.jar:?]
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74) [fabric-loader-0.14.14.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23) [fabric-loader-0.14.14.jar:?]
	at net.fabricmc.devlaunchinjector.Main.main(Main.java:86) [dev-launch-injector-0.2.1+build.8.jar:?]
Caused by: java.lang.RuntimeException: Mixin transformation of net.minecraft.world.entity.player.Player failed
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.getPostMixinClassByteArray(KnotClassDelegate.java:427) ~[fabric-loader-0.14.14.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.tryLoadClass(KnotClassDelegate.java:323) ~[fabric-loader-0.14.14.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.loadClass(KnotClassDelegate.java:218) ~[fabric-loader-0.14.14.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassLoader.loadClass(KnotClassLoader.java:112) ~[fabric-loader-0.14.14.jar:?]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:520) ~[?:?]
	at net.minecraft.client.main.Main.main(Main.java:136) ~[minecraft-merged-1991f807a5-1.18.2-loom.mappings.1_18_2.layered+hash.2198-v2.jar:?]
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:461) ~[fabric-loader-0.14.14.jar:?]
	... 3 more
Caused by: org.spongepowered.asm.mixin.transformer.throwables.MixinTransformerError: An unexpected critical error was encountered
	at org.spongepowered.asm.mixin.transformer.MixinProcessor.applyMixins(MixinProcessor.java:392) ~[sponge-mixin-0.11.4+mixin.0.8.5.jar:0.11.4+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClass(MixinTransformer.java:234) ~[sponge-mixin-0.11.4+mixin.0.8.5.jar:0.11.4+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClassBytes(MixinTransformer.java:202) ~[sponge-mixin-0.11.4+mixin.0.8.5.jar:0.11.4+mixin.0.8.5]
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.getPostMixinClassByteArray(KnotClassDelegate.java:422) ~[fabric-loader-0.14.14.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.tryLoadClass(KnotClassDelegate.java:323) ~[fabric-loader-0.14.14.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.loadClass(KnotClassDelegate.java:218) ~[fabric-loader-0.14.14.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassLoader.loadClass(KnotClassLoader.java:112) ~[fabric-loader-0.14.14.jar:?]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:520) ~[?:?]
	at net.minecraft.client.main.Main.main(Main.java:136) ~[minecraft-merged-1991f807a5-1.18.2-loom.mappings.1_18_2.layered+hash.2198-v2.jar:?]
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:461) ~[fabric-loader-0.14.14.jar:?]
	... 3 more
Caused by: java.lang.NoClassDefFoundError: com/llamalad7/mixinextras/injector/ModifyExpressionValue
	at com.bawnorton.mixinsquared.adjuster.tools.AdjustableAnnotationNode$KnownAnnotations.<clinit>(AdjustableAnnotationNode.java:116) ~[mixinsquared-fabric-0.2.0-beta.4.jar:?]
	at com.bawnorton.mixinsquared.adjuster.tools.AdjustableAnnotationNode.fromNode(AdjustableAnnotationNode.java:65) ~[mixinsquared-fabric-0.2.0-beta.4.jar:?]
	at com.bawnorton.mixinsquared.adjuster.ExtensionAnnotationAdjust.lambda$preApply$2(ExtensionAnnotationAdjust.java:71) ~[mixinsquared-fabric-0.2.0-beta.4.jar:?]
	at java.util.ArrayList.forEach(ArrayList.java:1511) ~[?:?]
	at com.bawnorton.mixinsquared.adjuster.ExtensionAnnotationAdjust.lambda$preApply$5(ExtensionAnnotationAdjust.java:64) ~[mixinsquared-fabric-0.2.0-beta.4.jar:?]
	at java.lang.Iterable.forEach(Iterable.java:75) ~[?:?]
	at com.bawnorton.mixinsquared.adjuster.ExtensionAnnotationAdjust.lambda$preApply$6(ExtensionAnnotationAdjust.java:58) ~[mixinsquared-fabric-0.2.0-beta.4.jar:?]
	at java.util.Optional.ifPresent(Optional.java:178) ~[?:?]
	at com.bawnorton.mixinsquared.adjuster.ExtensionAnnotationAdjust.preApply(ExtensionAnnotationAdjust.java:56) ~[mixinsquared-fabric-0.2.0-beta.4.jar:?]
	at org.spongepowered.asm.mixin.transformer.ext.Extensions.preApply(Extensions.java:156) ~[sponge-mixin-0.11.4+mixin.0.8.5.jar:0.11.4+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.TargetClassContext.preApply(TargetClassContext.java:413) ~[sponge-mixin-0.11.4+mixin.0.8.5.jar:0.11.4+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.TargetClassContext.applyMixins(TargetClassContext.java:401) ~[sponge-mixin-0.11.4+mixin.0.8.5.jar:0.11.4+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.MixinProcessor.applyMixins(MixinProcessor.java:363) ~[sponge-mixin-0.11.4+mixin.0.8.5.jar:0.11.4+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClass(MixinTransformer.java:234) ~[sponge-mixin-0.11.4+mixin.0.8.5.jar:0.11.4+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClassBytes(MixinTransformer.java:202) ~[sponge-mixin-0.11.4+mixin.0.8.5.jar:0.11.4+mixin.0.8.5]
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.getPostMixinClassByteArray(KnotClassDelegate.java:422) ~[fabric-loader-0.14.14.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.tryLoadClass(KnotClassDelegate.java:323) ~[fabric-loader-0.14.14.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.loadClass(KnotClassDelegate.java:218) ~[fabric-loader-0.14.14.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassLoader.loadClass(KnotClassLoader.java:112) ~[fabric-loader-0.14.14.jar:?]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:520) ~[?:?]
	at net.minecraft.client.main.Main.main(Main.java:136) ~[minecraft-merged-1991f807a5-1.18.2-loom.mappings.1_18_2.layered+hash.2198-v2.jar:?]
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:461) ~[fabric-loader-0.14.14.jar:?]
	... 3 more
Caused by: java.lang.ClassNotFoundException: com.llamalad7.mixinextras.injector.ModifyExpressionValue
	at jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:641) ~[?:?]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:520) ~[?:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.loadClass(KnotClassDelegate.java:226) ~[fabric-loader-0.14.14.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassLoader.loadClass(KnotClassLoader.java:112) ~[fabric-loader-0.14.14.jar:?]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:520) ~[?:?]
	at com.bawnorton.mixinsquared.adjuster.tools.AdjustableAnnotationNode$KnownAnnotations.<clinit>(AdjustableAnnotationNode.java:116) ~[mixinsquared-fabric-0.2.0-beta.4.jar:?]
	at com.bawnorton.mixinsquared.adjuster.tools.AdjustableAnnotationNode.fromNode(AdjustableAnnotationNode.java:65) ~[mixinsquared-fabric-0.2.0-beta.4.jar:?]
	at com.bawnorton.mixinsquared.adjuster.ExtensionAnnotationAdjust.lambda$preApply$2(ExtensionAnnotationAdjust.java:71) ~[mixinsquared-fabric-0.2.0-beta.4.jar:?]
	at java.util.ArrayList.forEach(ArrayList.java:1511) ~[?:?]
	at com.bawnorton.mixinsquared.adjuster.ExtensionAnnotationAdjust.lambda$preApply$5(ExtensionAnnotationAdjust.java:64) ~[mixinsquared-fabric-0.2.0-beta.4.jar:?]
	at java.lang.Iterable.forEach(Iterable.java:75) ~[?:?]
	at com.bawnorton.mixinsquared.adjuster.ExtensionAnnotationAdjust.lambda$preApply$6(ExtensionAnnotationAdjust.java:58) ~[mixinsquared-fabric-0.2.0-beta.4.jar:?]
	at java.util.Optional.ifPresent(Optional.java:178) ~[?:?]
	at com.bawnorton.mixinsquared.adjuster.ExtensionAnnotationAdjust.preApply(ExtensionAnnotationAdjust.java:56) ~[mixinsquared-fabric-0.2.0-beta.4.jar:?]
	at org.spongepowered.asm.mixin.transformer.ext.Extensions.preApply(Extensions.java:156) ~[sponge-mixin-0.11.4+mixin.0.8.5.jar:0.11.4+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.TargetClassContext.preApply(TargetClassContext.java:413) ~[sponge-mixin-0.11.4+mixin.0.8.5.jar:0.11.4+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.TargetClassContext.applyMixins(TargetClassContext.java:401) ~[sponge-mixin-0.11.4+mixin.0.8.5.jar:0.11.4+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.MixinProcessor.applyMixins(MixinProcessor.java:363) ~[sponge-mixin-0.11.4+mixin.0.8.5.jar:0.11.4+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClass(MixinTransformer.java:234) ~[sponge-mixin-0.11.4+mixin.0.8.5.jar:0.11.4+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClassBytes(MixinTransformer.java:202) ~[sponge-mixin-0.11.4+mixin.0.8.5.jar:0.11.4+mixin.0.8.5]
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.getPostMixinClassByteArray(KnotClassDelegate.java:422) ~[fabric-loader-0.14.14.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.tryLoadClass(KnotClassDelegate.java:323) ~[fabric-loader-0.14.14.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.loadClass(KnotClassDelegate.java:218) ~[fabric-loader-0.14.14.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassLoader.loadClass(KnotClassLoader.java:112) ~[fabric-loader-0.14.14.jar:?]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:520) ~[?:?]
	at net.minecraft.client.main.Main.main(Main.java:136) ~[minecraft-merged-1991f807a5-1.18.2-loom.mappings.1_18_2.layered+hash.2198-v2.jar:?]
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:461) ~[fabric-loader-0.14.14.jar:?]
	... 3 more

Process finished with exit code 1
```

I'm using:
```
minecraft_version = 1.18.2
fabric_version = 0.77.0+1.18.2
fabric_loader_version = 0.14.14
```

Can both problems be solved? How?  
Thanks in advance

**Comment by Crystal-Spider** [comment]
Note: if I use Mixin Squared 0.1.1 I don't get the runtime error, but I still need to add jitpack repo in B

**Comment by Bawnorton** [comment]
The dependency issue is a loom thing as loom strips transitive dependencies. Cant do anything about that; the same applies for any lib. The other issue is an oversight of expecting mixin extras to be present, I can fix that shortly.

**Comment by Bawnorton** [comment]
Mixin extras issue has been fixed, release will be out shortly as `0.2.0-beta.5`
